### PR TITLE
Reportback form cropping

### PR DESCRIPTION
**Pull request by sbsmith86**
@aaronschachter Wanted to send a pull request so you can start check out some of the module level code I added. 

:warning:  Most of this code (especially the js) is not production ready!! Just want to start getting eyes on this so we can move it to staging to start testing on devices.

### CODE REVIEW

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc [hunk 1]
```diff
@@ -243,9 +256,28 @@ function dosomething_reportback_form_save_file(&$form_state) {
...
+  $crop_y = $form_state['values']['crop_y'];
+  $crop_width = $form_state['values']['crop_width'];
+  $crop_height = $form_state['values']['crop_height'];
+  // Crop the image.
+  dosomething_reportback_image_crop($file,$crop_x,$crop_y,$crop_width,$crop_height);
   return $file;
 }
 
+// Function to crop image.
+function dosomething_reportback_image_crop($file,$crop_x,$crop_y,$crop_width,$crop_height) {
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc]: I'd recommend just using the same params as `image_crop` to make this a little more readable.  Also, params should have spaces.

```
function image_crop(stdClass $image, $x, $y, $width, $height)
```

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc [hunk 2]
```diff
@@ -243,9 +256,28 @@ function dosomething_reportback_form_save_file(&$form_state) {
...
   $file->status = FILE_STATUS_PERMANENT;
   file_save($file);
+  // @TODO - only if cropping is active.
+  // Grab cropping information.
+  $crop_x = $form_state['values']['crop_x'];
+  $crop_y = $form_state['values']['crop_y'];
+  $crop_width = $form_state['values']['crop_width'];
+  $crop_height = $form_state['values']['crop_height'];
+  // Crop the image.
+  dosomething_reportback_image_crop($file,$crop_x,$crop_y,$crop_width,$crop_height);
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc]: Each parameter should have a space after each comma. https://www.drupal.org/coding-standards#functcall

**Comment by aaronschachter** [comment]
PHP is ok except for minor comments. We just need to add a new variable into the DS Reportback admin form to conditionally load the JS / set the crop elements / crop.  We could potentially set this variable in the DS Campaign admin form since there isnt one for Reportbacks, but we should probalby just create a different Reportback adminc onfig form.

**Comment by sbsmith86** [comment]
@weerd @DFurnes Do you guys want to take a look at some of the fender-y stuff? I tried to clean it up a little.

**Comment by aaronschachter** [comment]
Just a reminder @sbsmith86 that we can't merge this until all cropping is conditionally displayed by a variable (see `variable_get` / `variable_set`)

**Comment by sbsmith86** [comment]
@aaronschachter yep! Will do. Trying to tackle search and 404 theming today to get it off my plate so I can fully focus on this.

**Comment by DFurnes** [comment]
Looking nice! :+1:

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc [hunk 3]
```diff
@@ -243,9 +256,28 @@ function dosomething_reportback_form_save_file(&$form_state) {
...
   return $file;
 }
 
+// Function to crop image.
+function dosomething_reportback_image_crop($image, $x, $y, $width, $height) {
+  // Crop the image.
+  $result = image_crop($image, $x, $y, $width, $height);
+  // Save the image.
+  // @TODO - this saves the new cropped image over the original one. Eventually we want to save it as a separate image.
+  image_save($image);
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc:292]: FYI #3454 adds 2 new columns: `options_processed` and `fid_processed`.  We'll want to store the x,y,width,and height as a serialized array within the `options_processed`, and the resulting fid in `fid_processed`.

**Comment by aaronschachter** [comment]
#3469 introduces a `dosomething_reportback_is_crop_enabled` variable we should use to check whether or not to present the crop widget / save a processed file.

**Comment by sbsmith86** [comment]
@aaronschachter sweet. Hoping to jump back into this today.

**Comment by sbsmith86** [comment]
I'm going to close this pull request since it is really outdated.

**Pull request closed without merging**